cmd/publishing-bot/config/rules.go

```diff
@@ -168,7 +168,7 @@ func validateGoVersions(rules *RepositoryRules) (errs []error) {
 // go versions don't follow semver. Examples:
 // 1. 1.15.0 is invalid, 1.15 is valid
 // 2. 1.15.0-rc.1 is invalid, 1.15rc1 is valid
-var goVerRegex = regexp.MustCompile(`^([0-9]+)\.([0-9]+)([\.1-9A-Za-z\-]*$)`)
+var goVerRegex = regexp.MustCompile(`^([0-9]+)\.([0-9]\w*)(\.[1-9]\d*\w*)*$`)
```

Comment:
relaxing the original to 0-9 seems simpler:
```suggestion
var goVerRegex = regexp.MustCompile(`^([0-9]+)\.([0-9]+)([\.0-9A-Za-z\-]*$)`)
```

Reply:
if apply that i got the following errors from the tests
```
=== RUN   TestValidateGoVersion
    /Users/cpanato/code/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:63: go version '0.9.0' is invalid, but got no error
    /Users/cpanato/code/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:63: go version '1.9.0' is invalid, but got no error
    /Users/cpanato/code/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:63: go version '1.15.0' is invalid, but got no error
    /Users/cpanato/code/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:63: go version '1.15.0-beta.1' is invalid, but got no error
    /Users/cpanato/code/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:63: go version '1.15.0-rc.1' is invalid, but got no error
    /Users/cpanato/code/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:63: go version '2.0-alpha1' is invalid, but got no error
    /Users/cpanato/code/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:63: go version '999.999.9999-beta.1' is invalid, but got no error
--- FAIL: TestValidateGoVersion (0.00s)
```

Reply:
the old regex would also allow plenty of other invalid names, like `1.15......`

the minimal change to unblock publishing seems ok to me

Reply:
Another option is `^([0-9]+)\.([0-9]+)\.?([1-9]+[0-9A-Za-z\-]*)$` - not ideal but would restrict most of the cases we care about and help unblock.

would still fail the following cases but I don't think we'd come across them in the near future. 


```
--- FAIL: TestValidateGoVersion (0.00s)
    /home/nikhita/go/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:58: go version check failed for valid version '0.0''
    /home/nikhita/go/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:58: go version check failed for valid version '0.0rc1''
    /home/nikhita/go/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:58: go version check failed for valid version '0.9''
    /home/nikhita/go/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:58: go version check failed for valid version '1.9''
    /home/nikhita/go/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:58: go version check failed for valid version '2.0''
    /home/nikhita/go/src/k8s.io/publishing-bot/cmd/publishing-bot/config/rules_test.go:58: go version check failed for valid version '2.0alpha1''
```